### PR TITLE
UX and UI refactoring

### PR DESCRIPTION
**Pull request by willcurry**

### CODE REVIEW

File: ConsoleGame.java [hunk 1]
```diff
@@ -61,4 +74,10 @@ public void gameIsOver(Board board, Player winner) {
         }
         print("\n" + winner.getMark() + " has won the game!\n");
     }
+
+    @Override
+    public String userPickGameMode() {
+        Scanner s = new Scanner(System.in);
```
Comment (sarahabimay) [inline ConsoleGame.java]: Shouldn't this be the _reader_ now?

File: HumanPlayer.java [hunk 2]
```diff
@@ -1,33 +1,24 @@
...
         int nextPosition = -1;
-        while (!validate(nextPosition)) {
-            try {
-                String line = inputReader.readLine();
-                nextPosition = Integer.parseInt(line);
-            } catch (Exception e) {
-                return nextPosition;
-            }
+        try {
+            nextPosition = Integer.parseInt(gameType.userInput());
```
Comment (sarahabimay) [inline HumanPlayer.java:15]: is gameType.userInput very descriptive to you?  I would make it a bit more explicit.  Also, is there an instruction to ask the user to enter a move?

File: Game.java [hunk 3]
```diff
@@ -58,12 +54,12 @@ public void playGame() {
...
         }
         gameType.gameIsOver(board, playerDeactive);
+        startNewGame();
     }
 
     public void pickGameMode() {
         gameType.displayAllGameModes();
-        Scanner s = new Scanner(System.in);
-        String input = s.nextLine();
+        String input = gameType.userPickGameMode();
```
Comment (sarahabimay) [inline Game.java]: The validation of the user input below, should also be in ConsoleGame.  Think: would it be necessary if you had a GUI or WebGUI?

File: Game.java [hunk 4]
```diff
@@ -75,11 +71,16 @@ public void pickGameMode() {
         pickGameMode();
     }
 
-    public static void main(String[] args) {
+    public static void startNewGame() {
         Board board = new Board("---------");
         Writer writer = new PrintWriter(System.out);
-        ConsoleGame consoleGame = new ConsoleGame(writer);
+        BufferedReader inputReader = new BufferedReader(new InputStreamReader(System.in));
```
Comment (sarahabimay) [inline Game.java]: this should be done in the Game's constructor..

File: GameTests.java [hunk 5]
```diff
@@ -30,8 +29,9 @@ public void playerMoveIsRegistered() {
     @Test
     public void multiplePlayerMovesAreRegistered() {
         InputStream stream = new ByteArrayInputStream("2\n3".getBytes());
+        BufferedReader inputReader = new BufferedReader(new InputStreamReader(stream));
```
Comment (sarahabimay) [inline GameTests.java]: Just saw that ConsoleGame constructor takes a BufferedReader but Game takes the InputStream and then constructs a BufferedReader in it's method: _startNewGame()_.  
I think these constructors should be consistent with each other e.g. either both take InputReader or both BufferedReader.  Probably InputReader as it is higher level than BufferedReader.

File: HumanPlayer.java (hunk 2 above)
Comment (willcurry) [inline HumanPlayer.java:15]: It is a bad name, but I cannot think of anything else and it tells them when its their turn.

File: ConsoleGame.java [hunk 6]
```diff
@@ -77,7 +73,18 @@ public void gameIsOver(Board board, Player winner) {
 
     @Override
     public String userPickGameMode() {
-        Scanner s = new Scanner(System.in);
-        return s.nextLine();
+        String input = null;
+        try {
+            input = reader.readLine();
+            for (Gamemodes gm : Gamemodes.values()) {
+                if (input.equals(gm.toString().toLowerCase()) || input.equals(gm.toString())) {
```
Comment (sarahabimay) [inline ConsoleGame.java:80]: Gamemodes should be GameModes.

File: HumanPlayer.java (hunk 2 above)
Comment (sarahabimay) [inline HumanPlayer.java:15]: how about _nextMove_ ? :)

File: ConsoleGame.java [hunk 7]
```diff
@@ -3,9 +3,13 @@
 public class ConsoleGame implements GameType {
 
     private final Writer writer;
+    private InputStream stream;
+    private BufferedReader reader;
 
-    public ConsoleGame(Writer writer) {
+    public ConsoleGame(Writer writer, InputStream stream) {
         this.writer = writer;
+        this.stream = stream;
```
Comment (sarahabimay) [inline ConsoleGame.java:11]: do you need this _stream_ as a member variable?  If it isn't used other than in constructor then lose it.

**Pull request merged**